src/os_xml/os_xml.c

```diff
@@ -502,12 +502,13 @@ static int _writememory(const char *str, XML_TYPE type, size_t size,
 
 static int _writecontent(const char *str, size_t size, unsigned int parent, OS_XML *_lxml)
 {
-    _lxml->ct[parent] = (char *)calloc(size, sizeof(char));
+    _lxml->ct[parent] = (char *)calloc(size + 1, sizeof(char));
```

Comment:
After searching current function callers, i found that every one uses it by setting size with strlen+1, so there's no need to allocate one more element

Reply:
In fact, i dont know why strdup is not used to do this